app/src/main/java/org/wikipedia/views/UserMentionEditText.kt

```diff
@@ -164,44 +146,7 @@ class UserMentionEditText : PlainPasteEditText {
         }
     }
 
-    private fun createUserNameSpan(spannable: Spannable, start: Int, end: Int) {
-        val span = UserColorSpan(ResourceUtil.getThemedColor(context, R.attr.progressive_color))
-        spannable.setSpan(span, start, end, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
-    }
-
-    /**
-     * Returns the text of this field, with properly expanded user mentions.
-     * Each user mention (in th form of @{username}) will be expanded into a wiki link to the
-     * user page of that user, i.e. [[User:username|@username]]
-     */
-    fun getParsedText(wikiSite: WikiSite): String {
-        var str = editable.toString()
-
-        val spans = editable.getSpans<UserColorSpan>()
-        if (spans.isNotEmpty()) {
-            val pairs = spans.map { MutablePair(editable.getSpanStart(it), editable.getSpanEnd(it)) }
-                .sortedBy { it.first }
-
-            for (i in pairs.indices) {
-                var name = str.substring(pairs[i].first, pairs[i].second)
-                if (name.length > 1 && name.startsWith("@")) {
-                    name = name.substring(1)
-                }
-                name = name.trim()
-                name = "[[" + UserAliasData.valueFor(wikiSite.languageCode) + ":" + name + "|@" + name + "]]"
-                str = str.replaceRange(pairs[i].first, pairs[i].second, name)
-
-                val lenDiff = name.length - (pairs[i].second - pairs[i].first)
-                for (j in i + 1 until pairs.size) {
-                    pairs[j].first += lenDiff
-                    pairs[j].second += lenDiff
-                }
-            }
-        }
-        return str
+    private fun composeUserNameLink(userName: String, wikiSite: WikiSite): String {
+        return "@[[" + UserAliasData.valueFor(wikiSite.languageCode) + ":" + userName + "|" + userName + "]]"
```

Comment:
It looks like the new @ mention text format will not include the `@` in the clickable link, is that expected?

Reply:
yes, it's expected, and that's actually how it looks in desktop DiscussionTools.